packages/nimble-components/src/table/models/table-layout-manager.ts

```diff
@@ -0,0 +1,358 @@
+        }
+        this.currentTotalDelta = this.pinColumnSizeDelta(
+            this.activeColumnDivider!,
+            this.currentTotalDelta
+        );
+        this.performCascadeSizeLeft(
+            this.activeColumnDivider!,
+            this.currentTotalDelta
+        );
+        this.performCascadeSizeRight(
+            this.activeColumnDivider!,
+            this.currentTotalDelta
+        );
+
+        this.table.tableWidthFactor = Math.max(
+            this.getCurrentTotalTableWidth() / this.initialTableWidth!,
+            1
+        );
+    };
+
+    private readonly onDividerMouseUp = (): void => {
+        document.removeEventListener('mousemove', this.onDividerMouseMove);
+        document.removeEventListener('mouseup', this.onDividerMouseUp);
+        this.unflagActiveColumnDividers();
+        this.updateTableViewportMinWidth();
+        this.resetGridSizedColumns();
+        this.table.isColumnBeingSized = false;
+    };
+
+    private getTotalColumnFixedWidth(): number {
+        return this.table.columns.reduce(
+            (accumulator: number, currentValue) => {
+                return (
+                    accumulator
+                    + (currentValue.columnInternals.currentPixelWidth
+                    !== undefined
+                        ? currentValue.columnInternals.currentPixelWidth
+                        : 0)
+                );
+            },
+            0
+        );
+    }
+
+    private getCurrentTotalTableWidth(): number {
+        return (
+            this.initialTableScrollableWidth!
+            + (this.getTotalColumnFixedWidth() - this.initialColumnTotalWidth!)
+        );
+    }
+
+    private setColumnsToFixedSize(): void {
+        this.cacheGridSizedColumns();
+        const headers = this.table.columnHeadersContainer.querySelectorAll(
+            '.header-container'
+        );
+        for (let i = 0; i < headers.length; i++) {
+            this.table.columns[i]!.columnInternals.currentPixelWidth = headers[i]!.getBoundingClientRect().width;
+        }
+        this.cacheColumnInitialPixelWidths();
+    }
+
+    private pinColumnSizeDelta(
+        activeColumnIndex: number,
+        delta: number
+    ): number {
+        let availableSpace = 0;
+        let currentIndex = activeColumnIndex;
+        if (delta > 0) {
+            // size right
+            availableSpace = delta;
+        } else if (delta < 0) {
+            // size left
+            while (currentIndex >= 0) {
+                const columnInitialWidths = this.initialColumnPixelWidths[currentIndex]!;
+                availableSpace
+                    += Math.floor(columnInitialWidths.initialPixelWidth)
+                    - columnInitialWidths.minPixelWidth;
+                currentIndex -= 1;
+            }
+        }
+        return delta > 0
+            ? Math.min(delta, availableSpace)
+            : Math.max(delta, -availableSpace);
+    }
+
+    private performCascadeSizeLeft(
```

Comment:
You will notice some obvious similarities between this method and `performCascadeSizeRight`. I made some attempts to refactor the methods to share some of the similar code, but didn't come up with something I liked. It's difficult because a lot of it is "reversed" logic. I still believe there is room to at least make some of the logic more consistent between the two, and am very interested in suggestions here.